src/security.ts

```diff
@@ -1,13 +1,19 @@
-import type { MiddlewareHandler } from "hono";
+import { createMiddleware } from "hono/factory";
+
+const https = "https:";
 
 /**
  * Enforce HTTPS connections.
  */
-export function httpsOnly(): MiddlewareHandler {
-	return async function middleware(context, next) {
-		let url = new URL(context.req.url);
-		if (url.protocol !== "http:") return await next();
```

Comment:
This check if it's not http: so other protocols also pass, only http: should be redirected to https:

Reply:
Ah, yes, my bad!

Reply:
I let you resolve this one if it's good for you?
Thanks for the review.